packages/purpose-process/src/services/purposeService.ts

```diff
@@ -323,6 +326,37 @@ export function purposeServiceBuilder(
         { readModelService, correlationId, repository }
       );
     },
+    async deletePurpose({
+      purposeId,
+      organizationId,
+      correlationId,
+    }: {
+      purposeId: PurposeId;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<void> {
+      logger.info(`Deleting Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+
+      assertOrganizationIsAConsumer(organizationId, purpose.data.consumerId);
+
+      assertPurposeIsDeletable(purpose.data);
+
+      const event = purposeIsDraft(purpose.data)
+        ? toCreateEventDraftPurposeDeleted({
+            purpose: purpose.data,
+            version: purpose.metadata.version,
+            correlationId,
+          })
+        : toCreateEventWaitingForApprovalPurposeDeleted({
+            purpose: purpose.data,
+            version: purpose.metadata.version,
+            correlationId,
+          });
+
+      await repository.createEvent(event);
```

Comment:
Let's wrap `event` in the `createEvent` when we don't need it outside

Reply:
```suggestion
      await repository.createEvent(
        purposeIsDraft(purpose.data)
          ? toCreateEventDraftPurposeDeleted({
              purpose: purpose.data,
              version: purpose.metadata.version,
              correlationId,
            })
          : toCreateEventWaitingForApprovalPurposeDeleted({
              purpose: purpose.data,
              version: purpose.metadata.version,
              correlationId,
            })
      );
```
This is what I meant, I don't think it would clutter much, it's still clear what the event is, but if you find it confusing we can leave it like you wrote it 👌🏻